easybuild/easyconfigs/p/Python/Python-2.7.13-foss-2016b-bare.eb

```diff
@@ -0,0 +1,147 @@
+name = 'Python'
+version = '2.7.13'
+versionsuffix = '-bare'
+
+homepage = 'http://python.org/'
+description = """Python is a programming language that lets you work more quickly and integrate your systems
+ more effectively.
+
+ This is a version without Tk or X11."""
+
+toolchain = {'name': 'foss', 'version': '2016b'}
+toolchainopts = {'pic': True}
+
+source_urls = ['http://www.python.org/ftp/%(namelower)s/%(version)s/']
+sources = [SOURCE_TGZ]
+
+# python needs bzip2 to build the bz2 package
+dependencies = [
+    ('bzip2', '1.0.6'),
+    ('zlib', '1.2.8'),
+    ('libreadline', '6.3'),
+    ('ncurses', '6.0'),
+    ('SQLite', '3.13.0'),
+    ('GMP', '6.1.2'),
+    ('libffi', '3.2.1'),
+    # OS dependency should be preferred if the os version is more recent then this version,
+    # it's nice to have an up to date openssl for security reasons
+    # ('OpenSSL', '1.0.2h'),
+]
+
+osdependencies = [('openssl-devel', 'libssl-dev', 'libopenssl-devel')]
+
+# order is important!
+# package versions updated 9th February 2017
+# using easy_update
+# the commented out extensions were added by easy_update but broke the build
+exts_list = [   
+    ('six', '1.10.0', {
+        'source_urls': ['https://pypi.python.org/packages/source/s/six/'],
+    }),
+    #('packaging', '16.8', {
+    #    'source_urls': ['https://pypi.python.org/packages/source/p/packaging/'],
+    #}),
+    #('pyparsing', '2.1.10', {
+    #    'source_urls': ['https://pypi.python.org/packages/source/p/pyparsing/'],
+    #}),
+    #('appdirs', '1.4.0', {
+    #    'source_urls': ['https://pypi.python.org/packages/source/a/appdirs/'],
+    #}),
```

Comment:
this looks related to recent changes in setuptools, where dependencies required by setuptools are not longer 'vendored' by setuptools and thus need to be installed first, but they require setuptools themselves... >_<

In Spack, this was fixed by patching the setuptools dependencies so they only rely on distutils rather than setuptools, cfr. https://github.com/LLNL/spack/pull/3198

Instead, I would rather use the most recent version of setuptools that doesn't require installing the dependencies first without having setuptools, i.e. setuptools version < 34.0, since setuptools is planning to take care of this problem themselves, see https://github.com/pypa/setuptools/issues/980